packages/openapi-generator/src/codec.ts

```diff
@@ -0,0 +1,286 @@
+    }
+    result.push(valueE.right);
+  }
+  return E.right({ type: 'tuple', schemas: result });
+}
+
+export function parsePlainInitializer(
+  project: Project,
+  source: SourceFile,
+  init: swc.Expression,
+): E.Either<string, Schema> {
+  if (init.type === 'ObjectExpression') {
+    return parseObjectExpression(project, source, init.span.start, init);
+  } else if (init.type === 'ArrayExpression') {
+    return parseArrayExpression(project, source, init);
+  } else if (init.type === 'StringLiteral') {
+    return E.right({ type: 'literal', kind: 'string', value: init.value });
+  } else if (init.type === 'NumericLiteral') {
+    return E.right({ type: 'literal', kind: 'number', value: init.value });
+  } else if (init.type === 'BooleanLiteral') {
+    return E.right({ type: 'literal', kind: 'boolean', value: init.value });
+  } else if (init.type === 'NullLiteral') {
+    return E.right({ type: 'literal', kind: 'null', value: null });
+  } else if (init.type === 'Identifier' && init.value === 'undefined') {
+    return E.right({ type: 'undefined' });
+  } else if (
+    init.type === 'Identifier' ||
+    init.type === 'MemberExpression' ||
+    init.type === 'CallExpression'
+  ) {
+    return parseCodecInitializer(project, source, init);
+  } else {
+    return E.right({ type: 'any' });
+  }
+}
+
+export function parseCodecInitializer(
+  project: Project,
+  source: SourceFile,
+  init: swc.Expression,
+): E.Either<string, Schema> {
+  if (init.type === 'Identifier' || init.type === 'MemberExpression') {
+    const identifierE = codecIdentifier(project, source, init);
+    if (E.isLeft(identifierE)) {
+      return identifierE;
+    }
+    const identifier = identifierE.right;
+
+    if (identifier.type === 'ref') {
+      return E.right(identifier);
+    }
+
+    const deref: DerefFn = () => E.left('Unimplemented deref');
+    return identifier.schema(deref);
+  } else if (init.type === 'CallExpression') {
+    const callee = init.callee;
+    if (callee.type !== 'Identifier' && callee.type !== 'MemberExpression') {
+      return E.left(`Unimplemented callee type ${init.callee.type}`);
+    }
+    const identifierE = codecIdentifier(project, source, callee);
+    if (E.isLeft(identifierE)) {
+      return identifierE;
+    }
+    const identifier = identifierE.right;
+
+    if (identifier.type === 'ref') {
+      return E.right(identifier);
+    }
+
+    function deref(source: SourceFile): DerefFn {
+      return (schema, fn) => {
+        if (schema.type !== 'ref') {
+          return fn(deref(source), schema);
+        } else {
+          const initE = findSymbolInitializer(project, source, schema.name);
+          if (E.isLeft(initE)) {
+            return E.left(initE.left);
+          }
+          const [newSourceFile, init] = initE.right;
+          const newSchema = parsePlainInitializer(project, newSourceFile, init);
+          if (E.isLeft(newSchema)) {
+            return E.left(newSchema.left);
+          }
```

Comment:
Nit: `newSchema` is either type; convention suggests `newSchema` should be named `newSchemaE`
```suggestion
          const newSchemaE = parsePlainInitializer(project, newSourceFile, init);
          if (E.isLeft(newSchemaE)) {
            return newSchemaE;
          }
```